src/lib/actions/helpers/decorators/__tests__/decorateHtmlCode.test.js

```diff
@@ -179,17 +183,20 @@ describe('decorate html code', function() {
       };
 
       var decorateHtmlCode = decorateHtmlCodeInjector();
-
-      decorateHtmlCode(
+      var onPromiseRejected = jasmine.createSpy('onPromiseRejected');
+      var decorateCodePromise = decorateHtmlCode(
         {
           settings: settings
         },
         settings.source
-      ).promise.catch(function() {
-        done();
-      });
+      ).promise;
+
+      decorateCodePromise.catch(onPromiseRejected).then(done);
```

Comment:
I think you can just do `decorateCodePromise.catch(fail)`. Also, if you were to return the promise from the test, this would be done automatically. For example:

```
return flushPromiseChains().then(function() {
  expect(onPromiseRejected).not.toHaveBeenCalled();
  window._satellite._onCustomCodeFailure(0);
  return decorateCodePromise;
});